File: src/main/java/tafqeetj/numbers/negative/NegativeQuantitativeNumberNamesFactory.java
package tafqeetj.numbers.negative;

import tafqeetj.numbers.QuantitiveNumberNames;
import tafqeetj.numbers.QuantitativeNumberNamesFactory;

public class NegativeQuantitativeNumberNamesFactory extends QuantitativeNumberNamesFactory {

    private static NegativeQuantitativeNumberNamesFactory instance;

    private NegativeQuantitativeNumberNamesFactory() {
    }

    public synchronized static NegativeQuantitativeNumberNamesFactory getInstance() {
        if (instance == null)
            instance = new NegativeQuantitativeNumberNamesFactory();
        return instance;
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/numbers/negative/NegativeQuantitativeNumberNamesFactory.kt
package tech.moussa.tafqeetk.numbers.negative

import tech.moussa.tafqeetk.numbers.QuantitativeNumberNamesFactory

class NegativeQuantitativeNumberNamesFactory private constructor(): QuantitativeNumberNamesFactory() {
    companion object {
        private var instance: NegativeQuantitativeNumberNamesFactory? = null

        @Synchronized
        fun getInstance(): NegativeQuantitativeNumberNamesFactory {
            if (NegativeQuantitativeNumberNamesFactory.instance == null) {
                NegativeQuantitativeNumberNamesFactory.instance =
                        NegativeQuantitativeNumberNamesFactory()
            }
            return NegativeQuantitativeNumberNamesFactory.instance!!
        }
    }
}